### PR TITLE
hw 24.1

### PR DESCRIPTION
**Pull request by mikhail771**

### CODE REVIEW

**Review by jah701 — APPROVED** [review]

File: src/main/java/Application.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import com.internet.mixshop.lib.Injector;
+import com.internet.mixshop.model.Product;
+import com.internet.mixshop.service.ProductService;
+
+public class Application {
```
Comment (KseniiaMakarova) [inline src/main/java/Application.java]: move this to `com.internet.mixshop` as well

File: src/main/java/com/internet/mixshop/dao/ProductDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package com.internet.mixshop.dao;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/dao/ProductDaoImpl.java]:
```suggestion
package com.internet.mixshop.dao.impl;
```

File: src/main/java/com/internet/mixshop/db/Storage.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+public class Storage {
+    public static final List<Product> products = new ArrayList<>();
+    public static final List<Order> orders = new ArrayList<>();
+    public static final List<ShoppingCart> shoppingCarts = new ArrayList<>();
+    public static final List<User> users = new ArrayList<>();
+    private static long productId = 0;
+    private static long orderId = 0;
+    private static long shoppingCartId = 0;
+    private static long userId = 0;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/db/Storage.java]:
```suggestion
    private static Long productId = 0L;
    private static Long orderId = 0L;
    private static Long shoppingCartId = 0L;
    private static Long userId = 0L;
```

File: src/main/java/com/internet/mixshop/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+public class Order {
+    private Long id;
+    private List<Product> products;
+    private Long userid;
+
+    public Order(List<Product> products, Long userid) {
+        this.id = id;
+        this.products = products;
+        this.userid = userid;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/model/Order.java]: `this.id = id;`
this doesn't make sense

also, you will not receive `products` from outside
let's initialize it as a new ArrayList

File: src/main/java/com/internet/mixshop/model/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+import java.util.List;
+
+public class ShoppingCart {
+    private long id;
+    private List<Product> products;
+    private Long userId;
+
+    public ShoppingCart(List<Product> products, Long userId) {
+        this.products = products;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/model/ShoppingCart.java]: same

File: src/main/java/com/internet/mixshop/model/ShoppingCart.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package com.internet.mixshop.model;
+
+import java.util.List;
+
+public class ShoppingCart {
+    private long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/model/ShoppingCart.java:7]:
```suggestion
    private Long id;
```

add getter and fix setter for this field

File: src/main/java/com/internet/mixshop/model/User.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package com.internet.mixshop.model;
+
+public class User {
+    private Long id;
+    private String name;
+    private String login;
+    private String password;
+
+    public User(Long id, String name, String login, String password) {
+        this.id = id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/model/User.java]: id will be set in storage

File: src/main/java/com/internet/mixshop/model/Product.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
+package com.internet.mixshop.model;
+
+public class Product {
+    private Long id;
+    private String name;
+    private double price;
+
+    public Product(String name, double price) {
+        this.id = id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/model/Product.java]: `this.id = id;` this line makes no sense

File: src/main/java/com/internet/mixshop/service/ProductServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
+package com.internet.mixshop.service;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/service/ProductServiceImpl.java]:
```suggestion
package com.internet.mixshop.service.impl;
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
also, add `<properties>` block and maven checkstyle plugin to pom.xml

use pom.xml from previous howeworks a reference

File: src/main/java/com/internet/mixshop/model/Order.java (hunk 4 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/model/Order.java]: done

File: src/main/java/com/internet/mixshop/dao/ProductDaoImpl.java (hunk 2 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/dao/ProductDaoImpl.java]: done

File: src/main/java/Application.java (hunk 1 above)
Comment (mikhail771) [reply src/main/java/Application.java]: done

File: src/main/java/com/internet/mixshop/db/Storage.java (hunk 3 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/db/Storage.java]: done

File: src/main/java/com/internet/mixshop/model/Order.java (hunk 4 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/model/Order.java]: done

File: src/main/java/com/internet/mixshop/model/Product.java (hunk 8 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/model/Product.java]: done

File: src/main/java/com/internet/mixshop/model/ShoppingCart.java (hunk 6 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/model/ShoppingCart.java:7]: done

File: src/main/java/com/internet/mixshop/model/ShoppingCart.java (hunk 5 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/model/ShoppingCart.java]: done

File: src/main/java/com/internet/mixshop/model/User.java (hunk 7 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/model/User.java]: done

File: src/main/java/com/internet/mixshop/service/ProductServiceImpl.java (hunk 9 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/service/ProductServiceImpl.java]: done

File: pom.xml [hunk 10]
```diff
@@ -7,6 +7,53 @@
...
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
 
 
 </project>
```
Comment (KseniiaMakarova) [inline pom.xml:59]:
```suggestion
</project>

```
add empty line in the end

File: src/main/java/com/internet/mixshop/model/ShoppingCart.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
+package com.internet.mixshop.model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShoppingCart {
+    private long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/mixshop/model/ShoppingCart.java:7]:
```suggestion
    private Long id;
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**